src/iter/collect/consumer.rs

```diff
@@ -13,54 +12,94 @@ pub(super) struct CollectConsumer<'c, T: Send> {
     /// certain amount of trust that the producer driving this will behave
     /// itself. Since this count is important to the safety of marking the
     /// memory initialized (`Vec::set_len`), we choose to keep it internal.
```

Comment:
This comment explains the previous qualms with this solution, yet I can't see anymore that it would be possible to "fake" the reduced CollectResult value.